### ISSUE TITLE
Is there any way to select dates directly from a calendar in swagger UI?

### ISSUE THREAD

**Issue by gerardcsaperas** [swagger-api/swagger-ui#6768]
I was wondering if there was any alternative to making the user write a date string in the parameters. :)

Thank you in advance!

Gerard

**Comment by mathis-m** [comment]
@gerardcsaperas yes this can be done by [providing your own component.](https://swagger.io/docs/open-source-tools/swagger-ui/customization/plugin-api/#components)
Parameter inputs are rendered by Specific components. Those are selected via this pattern:
`JsonSchema_${type}_${format}`  > `JsonSchema_${type}` > `JsonSchema_string`
Just add a React component via the Plugin API with the name `JsonSchema_string_date`.

For examples look at the [default JsonSchema Components](https://github.com/swagger-api/swagger-ui/blob/master/src/core/json-schema-components.jsx#L63).

Would be nice to see the results!

**Comment by s-bauer** [comment]
@gerardcsaperas This is a small example that demonstrates how to use the PluginAPI to render a custom component, in this case the date and date-time picker: https://gist.github.com/s-bauer/7ecfaf62c24a17c6a92c8599f414de37

There seems to be an issue with SwaggerUI tho. If I do provide an example for the query parameter in the swagger.yaml, then the value is the string literal "{}", which makes no sense.
